src/binary/cursor.rs

```diff
@@ -1216,8 +1238,41 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_read_decimal_zero() -> IonResult<()> {
+        let mut cursor = ion_cursor_for(&[0x50]);
+        assert_eq!(cursor.next()?, Some(Value(IonType::Decimal, false)));
+        assert_eq!(cursor.read_decimal()?, Some(Decimal::new(0, 0)));
+        Ok(())
+    }
+
+    #[test]
+    fn test_read_decimal_negative_zero() -> IonResult<()> {
+        let mut cursor = ion_cursor_for(&[0x52, 0x80, 0x80]);
+        assert_eq!(cursor.next()?, Some(Value(IonType::Decimal, false)));
+        assert_eq!(cursor.read_decimal()?, Some(Decimal::negative_zero()));
+        Ok(())
+    }
+
+    #[test]
+    fn test_read_decimal_positive_exponent() -> IonResult<()> {
+        let mut cursor = ion_cursor_for(&[0x52, 0x81, 0x02]);
+        assert_eq!(cursor.next()?, Some(Value(IonType::Decimal, false)));
+        assert_eq!(cursor.read_big_decimal()?, Some(20.into()));
```

Comment:
Allow deprecated missing here:
```suggestion
        #![allow(deprecated)] // `read_big_decimal` is deprecated
        assert_eq!(cursor.read_big_decimal()?, Some(20.into()));
```

Reply:
Good catch! This was a copy/paste error; it should be `read_decimal`. Pushing a fix now.